api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -173,22 +203,25 @@ public void cleanBillingBuffer() {
         .filter(entry -> entry.getLastSyncRequestTime() != null)
         .filter(
             entry ->
-                Duration.between(
-                            entry.getLastStatusChangedTime().toInstant(),
-                            entry.getLastSyncRequestTime().toInstant())
-                        .toMillis()
-                    >= CREATING_TIMEOUT.toMillis())
+                Comparables.isLessThan(
+                    CREATING_TIMEOUT, entry.getLastChangedToLastSyncRequestInterval()))
         .collect(Collectors.toList());
   }
 
-  public DbBillingProjectBufferEntry assignBillingProject(DbUser user) {
+  private Duration getStatusChangedToLastSync(DbBillingProjectBufferEntry entry) {
+    return Duration.between(
+        entry.getLastStatusChangedTime().toInstant(), entry.getLastSyncRequestTime().toInstant());
+  }
+
+  public DbBillingProjectBufferEntry assignBillingProject(DbUser dbUser) {
     DbBillingProjectBufferEntry entry = consumeBufferEntryForAssignment();
```

Comment:
Opt Nit: Not actually part of this PR, but all these objects named `entry` should probably be fully qualified imo. Just to make it clear what types we're working with?

Reply:
I'm pretty verbose, but if we name them all `dbBillingProjectBufferEntry`, it's a mouthful. I'll change them to `bufferEntry`. Should be clear enough, since this is the only buffer we really have in play here.